src/PhpDoc/PhpDocNodeResolver.php

```diff
@@ -373,6 +377,29 @@ public function resolveIsFinal(PhpDocNode $phpDocNode): bool
 		return count($finalTags) > 0;
 	}
 
+	/**
+	 * @return array<FriendTag>
+	 */
+	public function resolveFriends(PhpDocNode $phpDocNode, NameScope $nameScope): array
+	{
+		return array_map(function (PhpDocTagNode $node) use ($nameScope): FriendTag {
+			$genericTagValue = $node->value;
+			if (!$genericTagValue instanceof GenericTagValueNode) {
+				throw new \PHPStan\ShouldNotHappenException(sprintf(
```

Comment:
Other tags are filtered to the correct node with array_filter., it doesn't crash on wrong behaviour